### ISSUE TITLE
Issues with Fixed Seeds, Optuna Study Resetting, and Duplicate Trial Detection

### ISSUE THREAD

**Issue by galatasaraycf** [kochlisGit/ProphitBet-Soccer-Bets-Predictor#109]
I’ve noticed that all models currently use a fixed random seed (random_state=0 or seed=0), which seems to result in identical outputs when the same parameters are used. Also, each time a Tuner is initialized, a new Optuna study is created, so previous trials aren’t retained. Because of this, duplicate detection doesn’t seem to work as expected, and the same parameter combinations are being retried.Would it be possible to consider using persistent study storage and dynamic seeding to help improve trial diversity and avoid redundant evaluations?

**Comment by kochlisGit** [comment]
Hello, that's a very interesting suggestion you are proposing. However, I am not really convinced about these changes due to the following reasons:

# 1. Random Seed

The seed should probably remain fixed, so that the model initialization is always the same! This way, you can easily find the best hyperparameters of the model, without having to worry about bad initialization issues (for instance, bad initial weights in XGBoost, Random Forest or even Neural Networks).

# 2. Optuna Study

I could store the study, but that would eventually take up too much space, especially if you have created many models and many leagues. To avoid that, i don't save the study, but i provide the option of tuning a model up to 1000 trials.

**Comment by lionfish56** [comment]
I use  TPE Sampler and pruning in Optuna, so i cannot see that always the same parameters are used.
